### PR TITLE
News refactor

### PR DESCRIPTION
**Pull request by ekes**
As discussed:
 * Merges newsroom and news into one module;
 * Deploys an initial newsroom ('News' on /news) by default;
 * Associates news articles with a newsroom (defaults to single newsroom);
 * If a second newsroom is added, newsroom field on news articles allows them to be added to a newsroom;
 * Lists on a newsroom are filtered by the news in that newsroom;
 * The lists are on the newsroom node - as are search and facets.

### CODE REVIEW

**Comment by ekes** [comment]
Will need a new issue (in the theme?) to do the same as 'directories' and 'events' to make a column for the search/facets.

**Comment by ekes** [comment]
Will want a follow-up issue for filtering the search by newsroom. This has been de-prioritised as the use case for multiple newsrooms isn't yet spec'd.

**Comment by willguv** [comment]
@ekes if I install a second newsroom, will it also install at /news by default? As a content designer, how do I change this? Thanks

(Nice work by the way)

**Comment by ekes** [comment]
> @ekes if I install a second newsroom, will it also install at /news by default? As a content designer, how do I change this? Thanks

So it installs a newsroom called 'News' that has the path '/news' and all content posted into it has the path '/news/YEAR/article-title'.

If you create another newsroom it with have (by default) the path created from the title you give it 'Blog' would be '/blog' and all content posted into it has the path '/blog/YEAR/article-title'. You can change the newsroom path in the same way that you can change the path of a service (landing page) or guide. This will then be reflected in articles that are posted into the newsroom from that point.

Equally if you want you can delete or change the first newsroom that was created and have it on /newsroom or whatever.

The second follow-up I mention. The search. That still all at /news/search as there is just one news search. There are more ways to do this (split the search to the newsrooms, have one search that filers by newsroom etc.) and it can be tackled once we have people with multiple newsrooms and a set of stories about having to search them.

File: src/FeaturedNewsHelper.php [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+   * that are 'Promoted to frontpage'.
+   * The limit on the featureD_news display pages is the maximum number.
+   * There are between 0 (if nothing referenced or promoted) and the maximum
+   * number in *** FEATURED ***.
+   * The list can be exclusive IN (the featured listed) and excluded NOT IN (the
+   * rest of the items list).
+   *
+   * @param Drupal\views\ViewExecutable $view
+   *   The 'localgov_news_list' view.
+   * @param Drupal\views\Plugin\views\query\QueryPluginBase $query
```
Comment (ekes) [inline src/FeaturedNewsHelper.php]: Sql it's been checked in the module file before calling this function.

**Review by danchamp — APPROVED** [review]

**Comment by ekes** [comment]
localgovdrupal/localgov_core#77 and #32 are the follow-ups that this doesn't address.

**Comment by ekes** [comment]
Do we need to pass against 2.x already?

Or can we merge this. It'd be nice to do #32 before tagging a stable at least anyway.

**Comment by ekes** [comment]
So merge this.
Make another PR for 2.x compatibility.
And potentially #32

**Pull request merged**